src/debugpy/launcher/handlers.py

```diff
@@ -38,6 +40,10 @@ def property_or_debug_option(prop_name, flag_name):
     python = request("python", json.array(str, size=(1,)))
     cmdline = list(python)
 
+    # see https://github.com/microsoft/debugpy/issues/861
+    if PYTHON_311:
+        cmdline += ["-X", "frozen_modules=off"]
+
```

Comment:
Given that the issue only arises when debugger has to deal with `__file__`, this should probably go under the conditional below (i.e. it shouldn't be applied for `"noDebug": true`).

Reply:
Test moved